lib/connection/pool.js

```diff
@@ -1134,6 +1142,16 @@ Pool.prototype.write = function(commands, options, cb) {
   // Get the requestId
   operation.requestId = commands[commands.length - 1].requestId;
 
+  if (supportsClusterTime(this.topology) && this.topology.clusterTime) {
+    commands.forEach(command => {
+      if (command instanceof Query) {
+        command.query.$clusterTime = this.topology.clusterTime;
```

Comment:
thought it was supposed to be a top-level field in the command? why is it different for queries?

Reply:
Also good catch.  Originally the spec was designed assuming existence of `OP_MSG` support, which the server is guaranteed to support with `maxWireVersion >= 6`.  Unfortunately, we will not necessarily have that done in time so we need to accommodate `OP_QUERY` _and_ `OP_MSG` forms, and the `OP_QUERY` form expects this field to exist within the top-level `query` field.  I've informed @rstam about this.